src/Form/Checkbox/Checkbox.component.tsx

```diff
@@ -0,0 +1,123 @@
+// REACT
+import * as React from 'react';
+const { forwardRef } = React;
+
+// VENDOR
+import styled, { css } from '@xstyled/styled-components';
+import { compose } from '@xstyled/system';
+import classnames from 'classnames';
+
+// ANCHOR
+import { Check } from '../..';
+import { StyledIcon } from '../../Icon/utils';
+import {
+    space as spaceStyles,
+    SpaceProps,
+    color as colorStyles,
+    ColorProps,
+    backgroundColor as backgroundColorStyles,
+    BackgroundColorProps,
+} from '../../theme/system.theme';
+
+const HiddenInput = styled('input')<CheckboxProps>`
+    visibility: hidden;
+    opacity: 0;
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+`;
+
+const StyledCheckbox = styled('label')<CheckboxProps>`
+    box-sizing: border-box;
+    position: relative;
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+
+    user-select: none;
+    border: dark;
+    border-radius: 2px;
+
+    ${compose(
+        spaceStyles,
+        colorStyles,
+        backgroundColorStyles
+    )}
+
+    ${({ size, disabled }) =>
+        css({
+            height: size,
+            width: size,
+            cursor: disabled ? 'default' : 'pointer',
+            opacity: disabled ? 0.4 : 1,
+        })}
+
+    ${StyledIcon} {
+        transition: opacity 100ms ease-in-out;
+        ${({ checked }) =>
+            css({
+                opacity: checked ? 1 : 0,
+            })}
+    }
+`;
+
+interface CheckboxProps extends SpaceProps, ColorProps, BackgroundColorProps {
```

Comment:
Given the event handlers, you may not be able to use `React.HTMLAttributes<HTMLInputElement>`, but can you verify that? I know AutoComplete has too many custom handlers to be compatible with that interface.

Reply:
Wasn't compatible, but how about:
```
    inputProps?: React.HTMLAttributes<HTMLInputElement>;
```
?

Reply:
This won't allow event binding on the parent checkbox, but it does solve a part of the issue. It's ok if this is omitted for now. I need to figure out a FormControl interface that allows us more flexibility.